model/db.js

```diff
@@ -29,7 +29,7 @@ var Notifications = new Schema({
 });
 
 var Challenges = new Schema({
-  name:           String,
+  name:           {type: String, unique : true},
```

Comment:
This is a validation at the database level, we also need to inform the end user about the error. Otherwise the application will yield 500 (perhaps?) and that's it.